libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClient.kt

```diff
@@ -477,19 +477,23 @@ class RustMatrixClient(
     override fun roomDirectoryService(): RoomDirectoryService = roomDirectoryService
 
     override fun close() {
+        innerNotificationClient.close()
+
         appCoroutineScope.launch {
             roomFactory.destroy()
             rustSyncService.destroy()
             notificationSettingsService.destroy()
+            // This is sync, but it can destroy the `Client` instance and block stopping the sync service
+            notificationProcessSetup.destroy()
         }
```

Comment:
Ideally all of these should be in a `runBlocking` block, but that breaks the `RustMatrixClient` related tests as soon as we reach a `withContext` call...